File: Generated_test_cases/cpp/src/directory_cleaner.cpp
#include "directory_cleaner.h"
#include <filesystem>

void recursiveDelete(const std::string& dirPath) {
    std::filesystem::path path(dirPath);
    if (!std::filesystem::exists(path)) return;
    for (const auto& entry : std::filesystem::directory_iterator(path)) {
        if (entry.is_directory()) {
            recursiveDelete(entry.path().string());
        } else {
            std::filesystem::remove(entry.path());
        }
    }
    std::filesystem::remove(path);
}

File: Generated_test_cases/java/src/mian/engine/cleaner/DirectoryCleaner.java
package engine.cleaner;

import java.io.IOException;
import java.nio.file.*;
import java.nio.file.attribute.BasicFileAttributes;

public class DirectoryCleaner {
    public static void recursiveDelete(String dirPath) throws IOException {
        Path path = Paths.get(dirPath);
        if (!Files.exists(path)) return;
        Files.walkFileTree(path, new SimpleFileVisitor<Path>() {
            @Override
            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
                Files.delete(file);
                return FileVisitResult.CONTINUE;
            }
            @Override
            public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
                Files.delete(dir);
                return FileVisitResult.CONTINUE;
            }
        });
    }
}